File: 030/030.rs
fn sum_of_digit_fifth_powers(n: u64) -> u64 {
    let sum: u64 = n.to_string().chars().map(|c| (c as u64 - 48).pow(5)).sum();
    return sum;
}

fn main() {
    let mut sum: u64 =  0;
    for n in 2..999999 {
        if n == sum_of_digit_fifth_powers(n) {
            sum += n;
            println!("{}", n);
        }
    }
    println!("{}", sum);
}

File: 001-050/030/030.rb
class Problem030
   def sum_of_digit_fifth_powers(n)
      n.to_s.chars.map {|s| (s.to_i)**5 }.sum
   end

   def compute()
     nums = (2..999999).select {|n| n == sum_of_digit_fifth_powers(n) }
     nums.sum
   end
end
p Problem030.new().compute()
